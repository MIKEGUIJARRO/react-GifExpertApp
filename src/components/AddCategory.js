import React, { useState } from "react";
import PropTypes from "prop-types";
//Aqui recibimos las (props) de AddCategory 
export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState("");

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setInputValue("");
            setCategories((items) => {
                return [inputValue, ...items];
            });
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input type="text" value={inputValue} onChange={handleInputChange} />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}
