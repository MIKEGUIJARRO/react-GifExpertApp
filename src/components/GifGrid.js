import React, { Fragment } from "react";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <Fragment>
            <h3>{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading ...</p>}
            <div className="card-grid">
                {
                    images.map((image) => {
                        return <GifGridItem key={image.id} {...image} />
                    })
                }
            </div>
        </Fragment>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}