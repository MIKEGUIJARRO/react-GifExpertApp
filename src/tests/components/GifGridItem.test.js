import { GifGridItem } from "../../components/GifGridItem";
import React from "react";
import { shallow } from "enzyme";



describe("<GifGridItem /> test", () => {
    const id = "12345";
    const title = "This is a title";
    const url = "https://localhost/demo.png";
    const wrapper = shallow(<GifGridItem id={id} title={title} url={url} />);

    test("Shows the <GifGridItem /> correctly", () => {

        expect(wrapper).toMatchSnapshot();
    });

    test("Has a paragraph within the title", () => {
        const p = wrapper.find("p");

        expect(p.text().trim()).toBe(title);
    });

    test("Should contain the same img as the URL and alt of props", () => {
        const img = wrapper.find("img");

        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(title);
    });

    test("Should contain animate__fadeIn class", () => {
        const div = wrapper.find("div");
        const classes = div.props().className;
        expect(classes.includes("animate__fadeIn")).toBe(true);
    });
});