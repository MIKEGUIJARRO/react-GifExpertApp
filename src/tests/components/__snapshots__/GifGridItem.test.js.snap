// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> test Shows the <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="This is a title"
    src="https://localhost/demo.png"
  />
  <p>
    This is a title
  </p>
</div>
`;
